synthtool/sources/git.py

```diff
@@ -45,26 +49,30 @@ def make_repo_clone_url(repo: str) -> str:
 
 
 def clone(
-    url: str,
-    dest: pathlib.Path = None,
-    committish: str = "master",
-    force: bool = False,
+    url: str, dest: pathlib.Path = None, committish: str = None, force: bool = False,
```

Comment:
Can we backfill some documentation here. It looks like we're changing the behavior of the default value of `committish`. It's now ignored if there's an existing clone specified. Otherwise, it defaults to "master"

Reply:
I added a complete docstring.

I don't see the path through the code where committish gets ignored.

Reply:
The default value for `committish` is only "master" for the non-preclone scenario. This is probably fine as if you are providing a pre-cloned repo, then presumably the current state of that directory acceptable

Reply:
Indeed, using the current state of that directory is required for the autosynth run-at-every-commit strategy to work.